packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -476,6 +476,52 @@ enum FloatingLabelBehavior {
   always,
 }
 
+/// Defines the horizontal alignment of a floating label within an
+/// [InputDecorator].
+///
+/// See also:
+///
+///  * [InputDecoration.floatingLabelAlignment] which defines the alignment for
+///    [InputDecoration.label] or [InputDecoration.labelText]
+///  * [FloatingLabelBehavior] which defines the behaviour of the floating label.
+@immutable
+class FloatingLabelAlignment {
+  const FloatingLabelAlignment._({
+    required this.x,
+  }) : assert(x != null),
+       assert(x >= -1.0 && x <= 1.0);
+
+  /// A value ranging from -1.0 to 1.0 inclusive, where -1.0 represents the
+  /// leftmost possible position, 1.0 represents the rightmost possible
+  /// position, and 0.0 represents the center position.
+  final double x;
+
+  /// Aligns a floating label to the leftmost possible position on top of
+  /// an input decorator.
```

Comment:
"on top of" => "in"

I think the InputDecorator really includes the space around the field, so the label is technically inside of it, not on top.

Also below for `center`.

Reply:
Also "input decorator" => "[InputDecorator]"